typing/types.mli

```diff
@@ -294,7 +294,7 @@ type type_declaration =
     (* definition level * expansion level *)
     type_loc: Location.t;
     type_attributes: Parsetree.attributes;
-    type_immediate: bool; (* true iff type should not be a pointer *)
+    type_repr: type_repr;
```

Comment:
We need to document this, I created an issue: to keep track: [MPR#7599](https://caml.inria.fr/mantis/view.php?id=7599).